packages/framework/immutable-merge/src/Merge.ts

```diff
@@ -117,6 +141,24 @@ function getHandlerForPropertyOfType(
   return result;
 }
 
+/**
+ * Assign properties of source objects to a new target object. This is just a type wrapper around Object.assign
+ * @param objs - array of objects to merge
+ * @returns the result of object assign on the objects, typed to T
+ */
+function assignToNewObject<T extends ObjectBase>(...objs: T[]): T {
+  return Object.assign({}, ...objs);
+}
+
+/**
+ * Filter a set of unknown values to only include those that extend ObjectBase, this actually
```

Comment:
was this comment meant to have more text?

Reply:
I think I was going to explain more about type inferencing but then decided that wasn't necessary. I'll fix it.